src/main/java/org/opensearch/security/dlic/rest/api/ActionGroupsApiAction.java

```diff
@@ -92,8 +93,35 @@ public List<Route> routes() {
     }
 
     @Override
-    protected AbstractConfigurationValidator getValidator(final RestRequest request, BytesReference ref, Object... param) {
-        return new ActionGroupValidator(request, isSuperAdmin(), ref, this.settings, param);
+    protected RequestContentValidator createValidator(final Object... params) {
+        return RequestContentValidator.of(new RequestContentValidator.ValidationContext() {
+            @Override
+            public Object[] params() {
+                return params;
+            }
+
+            @Override
+            public Settings settings() {
+                return settings;
+            }
+
+            @Override
+            public Map<String, RequestContentValidator.DataType> allowedKeys() {
+                final ImmutableMap.Builder<String, DataType> allowedKeys = ImmutableMap.builder();
+                if (isSuperAdmin()) {
+                    allowedKeys.put("reserved", DataType.BOOLEAN);
+                }
+                allowedKeys.put("allowed_actions", DataType.ARRAY);
```

Comment:
Does this go beyond defining an array? i.e. an array of action names (strings) vs an array of booleans

Reply:
No it doesn't and same for DataType.OBJECT. I noticed it as well.